src/megaapi_impl.cpp

```diff
@@ -23829,14 +23890,73 @@ void MegaFolderUploadController::cancel()
     //remove subtransfers from pending transferQueue
     megaApi->cancelPendingTransfersByFolderTag(tag);
 
+    //remove ongoing subtransfers
+    long long cancelledSubTransfers = 0;
+    DBTableTransactionCommitter committer(client->tctable);
     while (!subTransfers.empty())
     {
         auto subTransfer = *subTransfers.begin();
-        subTransfer->setState(MegaTransfer::STATE_COMPLETED);
-        DBTableTransactionCommitter committer(client->tctable);
-        megaApi->fireOnTransferFinish(subTransfer, make_unique<MegaErrorPrivate>(API_EINCOMPLETE), committer);
+
+        Transfer *transfer = subTransfer->getTransfer();
+        if (!transfer)
+        {
+            LOG_warn << "Subtransfer without attached Transfer for folder transfer: " << subTransfer->getFileName();
+
+            subTransfer->setState(MegaTransfer::STATE_COMPLETED); //TODO: copmleted?
+            megaApi->fireOnTransferFinish(subTransfer, make_unique<MegaErrorPrivate>(API_EINCOMPLETE), committer);
+
+            continue;
+        }
+
+#ifdef _WIN32
+        if (transfer->type == GET)
+        {
+            transfer->localfilename.append("", 1);
+            WIN32_FILE_ATTRIBUTE_DATA fad;
+            if (GetFileAttributesExW((LPCWSTR)transfer->localfilename.data(), GetFileExInfoStandard, &fad))
+                SetFileAttributesW((LPCWSTR)transfer->localfilename.data(), fad.dwFileAttributes & ~FILE_ATTRIBUTE_HIDDEN);
+            transfer->localfilename.resize(transfer->localfilename.size()-1);
```

Comment:
It looks like this is the same code as in MegaFileGet::updatelocalname() - we could call that once we find the File* inside the transfer instead?

Reply:
I took this code from TYPE_CANCEL_TRANSFER . I thought the idea was to unhide the temporary download file ... Not sure if this relates to File* ... :thinking:

Reply:
On second though, both cases could be wrong, and this should only be done if there were no more File* associated with that Transfer?

Reply:
I just tested the normal case, and the temp file for download is hidden on the first progress() call, and unhidden once all the data has been downloaded, in Transfer::complete().   I think this is probably just to keep the directory looking tidy if many files are being downloaded at once.   And for a cancelled download, we should be about to delete that temporary file anyway, so I think this is fine for now.   thanks